packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts

```diff
@@ -728,6 +729,10 @@ export class PluginViewRegistry implements FrontendApplicationContribution {
         }
     }
 
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    private shouldUpdateViewVisibility(clauseContext: any, e: any): boolean {
+        return clauseContext === undefined || clauseContext === 'true' || (clauseContext instanceof Set && e.affects(clauseContext));
```

Comment:
It seems that an alternative to the change here would be to change the `.affects()` method (and perhaps the `.parseKeys()` method used above) so that they can just handle `'true'` and `'false'` correctly, without special-casing them in this logic - though it might end up with just special casing them in that logic. Do you have any sense of costs or benefits to the two approaches?

Reply:
1. I agree this isn't the cleanest solution. I contemplated other options of just using the `Set<string>` or adding `boolean` but in the end it seemed redundant. 
2. Also, `affects` invokes in VS Code the [affectsome](https://github.com/microsoft/vscode/blob/aaf6287ecccebda0eba6f78c86670aeb3d8dfe22/src/vs/platform/contextkey/browser/contextKeyService.ts#L216) function - so can't change implementation there.

```
class SimpleContextKeyChangeEvent implements IContextKeyChangeEvent {
	constructor(readonly key: string) { }
	affectsSome(keys: IReadableSet<string>): boolean {
		return keys.has(this.key);
	}
}
```

3. I also didn't like that "truthy" check would be to check viewClauseContexts contains one item == 'true'
4. In my proposal solution, I'm referring to the entire contents of `view.when` - if it equals "true". I don't depend exception of some sort in parsing of keys or otherwise that would cause viewClauseContexts to contain 1 item == 'true' when it shouldn't have. 
5. And finally, this is also a temporary fix as in from `standalone/0.21.x` version, VS Code added `ContextKeyFalseExpr | ContextKeyTrueExpr` types which handle False/True expressions  (Theia currently uses only one `ContextKeyExpr` containing string[]). There is also currently a PR for theia/monaco-editor-core@0.23 and so after it - we could maybe refactor code to have more ContextKeyExpr types.
![image](https://user-images.githubusercontent.com/71069170/110493936-c24e7380-80fb-11eb-959b-8e2b32bee831.png)
https://github.com/microsoft/vscode/blob/b251bd952b84a3bdf68dad0141c37137dac55d64/src/vs/platform/contextkey/common/contextkey.ts#L57